spec/classes/init_spec.rb

```diff
@@ -1962,6 +1973,45 @@
     end
   end
 
+  describe 'with parameter manage_global_known_hosts' do
+    ['YES','badvalue',2.42,['array'],a = { 'ha' => 'sh' }].each do |value|
+      context "specified as invalid value #{value} (as #{value.class})" do
+        let(:params) { { :manage_global_known_hosts => value } }
+        it do
+          expect {
+            should contain_class('ssh')
+          }.to raise_error(Puppet::Error,/(is not a boolean|Unknown type of boolean)/)
+        end
+      end
+    end
+
+    ['true', true].each do |value|
+      context "specified as valid true value #{value} (as #{value.class})" do
+        let(:params) { { :manage_global_known_hosts => value } }
+        let(:params) { { :ssh_config_global_known_hosts_file => '/valid/path' } }
```

Comment:
since we are testing the values of manage_global_known_hosts and not ssh_config_global_known_hosts_file, ssh_config_global_known_hosts_file should not be specified. Please remove that line and use the default path instead of `/valid/path`